### DISCUSSION TITLE
Issue with is_dumb_terminal

### DISCUSSION THREAD

**Discussion by ronf** [Textualize/rich#1219] [Q&A]
The implementation of is_dumb_terminal in Rich currently depends on the TERM environment variable, but this isn't a good idea when using a console which is not associated the user's main TTY. Specifically, the current code looks like:

```
    @property
    def is_dumb_terminal(self) -> bool:
        """Detect dumb terminal.

        Returns:
            bool: True if writing to a dumb terminal, otherwise False.

        """
        _term = self._environ.get("TERM", "")
        is_dumb = _term.lower() in ("dumb", "unknown")
        return self.is_terminal and is_dumb
```

In my application, I have multiple clients connected to a shared process coming in over SSH, and each of them has its own separate Console object. I pass in force_terminal and color_system set based on the terminal type I get from SSH, but I don't have a good way to control whether is_dumb_terminal is set or not, since that depends on $TERM.

It looks like there's now something called is_interactive which I might be able to use in place of is_dumb_terminal, since it has a corresponding force_interactive variable I could set explicitly based on the SSH terminal type. However, there's still a place in the Live code (in refresh()) where it checks is_dumb_terminal, so I'm not sure this will actually work for me.

Would it be possible to use is_interactive consistently in Live rather than is_dumb_terminal, or provide some other way to force the setting of is_dumb_terminal when specifying your own Console?

**Comment by willmcgugan** [comment]
Perhaps there should be a `force_dumb` argument on the constructor. I'll think about it.

In the meantime, you could override `is_dumb_terminal` to return False. Alternatively, you can use the undocumented `_environ` argument to replace the environment variable mapping for a Console.
